lib_eio/process.ml

```diff
@@ -0,0 +1,31 @@
+type status = Exited of int | Signaled of int | Stopped of int
```

Comment:
Would something like 

```ocaml
type exit_status = [ `Exited of int | `Signaled of int ]

type status = [ exit_status | `Stopped of int ] 
```

be useful? If the `Stopped` case can sometimes happen in the exit case the `exit_status` method would still have to handle it. If it can never happen I suppose we can `assert false` it?